WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -38,21 +45,352 @@ class WordPressAuthenticatorTests: XCTestCase {
+        
+        guard let navController = presenterSpy.presentedVC as? LoginNavigationController, let controller = navController.viewControllers.first as? LoginEmailViewController else {
+            XCTFail("Could not fetch correct ViewController")
+            return
+        }
+        
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertEqual(controller.loginFields.restrictToWPCom, true)
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogXMLRPC, "https://example.com/xmlrpc.php")
+        XCTAssertEqual(controller.loginFields.meta.jetpackBlogUsername, "username")
+        XCTAssertEqual(controller.loginFields.username, "email-address@example.com")
+    }
+    
+    func testShowLoginForSelfHostedSitePresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+    
+    func testShowLoginForSelfHostedSiteTracksOpenLogin() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let delegateSpy = WordPressAuthenticatorDelegateSpy()
+        
+        WordPressAuthenticator.shared.delegate = delegateSpy
+        
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+        
+        WordPressAuthenticator.showLoginForSelfHostedSite(presenterSpy)
+        wait(for: [expectation], timeout: 3)
+        
+        let trackedEvent = delegateSpy.trackedElement
+        
+        XCTAssertEqual(trackedEvent, WPAnalyticsStat.openedLogin)
+    }
+    
+    func testSignInForWPComReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPCom()
+        
+        XCTAssertTrue((vc as Any) is LoginEmailViewController)
+    }
+    
+    func testSignInForWPComWithLoginFieldsReturnsVC() {
+        let navController = WordPressAuthenticator.signinForWPCom(dotcomEmailAddress: "example@email.com", dotcomUsername: "username") as! UINavigationController
+        let vc = navController.topViewController
+        
+        XCTAssertTrue((navController as Any) is UIViewController)
+        XCTAssertTrue((vc as Any) is LoginWPComViewController)
+    }
+    
+    func testSignInForWPComSetsEmptyLoginFields() {
+        let navController = WordPressAuthenticator.signinForWPCom(dotcomEmailAddress: nil, dotcomUsername: nil) as! UINavigationController
+        let vc = navController.topViewController as! LoginWPComViewController
+        
+        XCTAssertEqual(vc.loginFields.emailAddress, String())
+        XCTAssertEqual(vc.loginFields.username, String())
```

Comment:
```suggestion
        XCTAssertEqual(vc.loginFields.emailAddress, "")
        XCTAssertEqual(vc.loginFields.username, "")
```

There's nothing _wrong_ with using `String()`, but I think `""` is clearer. It takes less time to read, too, and the easier we can make our code to read, the easier it will be understand.

Reply:
Hmmm, I could have sworn that when I first wrote this test I user "" instead of String() and the test failed unless I initialized a string this way...  anyways....  I agree I think it is clearer, not sure why I thought this failed.  Updated